test/cql-pytest/test_permissions.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2021-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# Tests for managing permissions
+#############################################################################
+
+import pytest
+from cassandra.protocol import SyntaxException, InvalidRequest
+from util import new_test_table
+
+# Test that granting permissions to various resources works for the default user.
+# This case does not include functions, because due to differences in implementation
+# the tests will diverge between Scylla and Cassandra (e.g. there's no common language)
+# to create a user-defined function in.
+# Marked cassandra_bug, because Cassandra allows granting a DESCRIBE permission
+# to a specific ROLE, in contradiction to its own documentation:
+# https://cassandra.apache.org/doc/latest/cassandra/cql/security.html#cql-permissions
+def test_grant_applicable_data_and_role_permissions(cql, test_keyspace, cassandra_bug):
+    schema = "a int primary key"
+    user = "cassandra"
+    with new_test_table(cql, test_keyspace, schema) as table:
+        # EXECUTE is not listed, as it only applies to functions, which aren't covered in this test case
+        all_permissions = set(['create', 'alter', 'drop', 'select', 'modify', 'authorize', 'describe'])
+        applicable_permissions = {
+            'all keyspaces': ['create', 'alter', 'drop', 'select', 'modify', 'authorize'],
+            f'keyspace {test_keyspace}': ['create', 'alter', 'drop', 'select', 'modify', 'authorize'],
+            f'table {table}': ['alter', 'drop', 'select', 'modify', 'authorize'],
+            'all roles': ['create', 'alter', 'drop', 'authorize', 'describe'],
+            f'role {user}': ['alter', 'drop', 'authorize'],
+        }
+        for resource, permissions in applicable_permissions.items():
+            # Applicable permissions can be legally granted
+            for permission in permissions:
+                cql.execute(f"GRANT {permission} ON {resource} TO {user}")
```

Comment:
Yes, it is, and indeed for more complicated tests I'll need to make sure that all permissions are restored. But for that I also suspect I'm going to need to create temporary users, because the default user `cassandra` is also a superuser, which makes certain tests nonsensical. For that, it's likely I'll need to extend the test suite a little, so that new sub-sessions can be created with new users, where their permissions can be validated.